overlord/snapstate/snapmgr.go

```diff
@@ -411,6 +442,25 @@ func (snapst *SnapState) LocalRevision() snap.Revision {
 	return local
 }
 
+// LocalComponentRevision returns the "latest" local revision for the compName
+// component. Local revisions start at -1 and are counted down. 0 will be
+// returned if no local revision for the component is found.
+func (snapst *SnapState) LocalComponentRevision(compName string) snap.Revision {
```

Comment:
The name of this function could maybe better imply it is "latest" revision of the local component. Reading the name now, it isn't clear that it is looking for the latest local component.

Reply:
Although I see now that this matches the pattern of the preexisting `SnapState.LocalRevision`, so matching that is probably more important.